Create dialog system that prompts user to save or discard changes

lonesymmetry opened the pull request:
Note: this PR depends on PR #31.

Create a dialog system that can be reused easily and use this system to prompt the user to save or discard changes to the open workstation when they try to open a new one, create a new one, or exit the application.

![image](https://user-images.githubusercontent.com/12960336/109461369-e385d480-7a16-11eb-9d46-ebbddfb1c1e3.png)

lonesymmetry committed:
Begin work on saving and loading workstations

lonesymmetry committed:
Check for unsaved changes when replacing the current workstation

Also grey out the Save button if the current workstation has not already been saved as a filename

lonesymmetry committed:
Switch to new Input System Package for active input handling

lonesymmetry committed:
Add comments to workstation import and export code

Also add missing changes to package manager settings

lonesymmetry committed:
Begin work on dialog system

lonesymmetry committed:
Update title screen scene to use new input system event system

lonesymmetry committed:
Finish save dialog

lonesymmetry committed:
Fix bug where dialog footer buttons wouldn't highlight on hover

lonesymmetry committed:
Check for unsaved changes when user tries to quit application

macovskym reviewed (changes requested):
Looks good for the most part. However, opening saved workstations does not seem to work any more.

Additionally, as I said on the prior PR, I think the save button should be grayed out for new workstations.

lonesymmetry committed:
Fix a bug where CheckUnsavedChanges wouldn't run the next action callback

Also, give the scene a chance to render before prompting the user to load a workstation from the title screen.
Lastly, also ensure that New is always called to set things up when the MainScene starts

macovskym reviewed (approved):
Nice! Everything seems to be working now.
